sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -411,6 +413,34 @@ class SparkSession private(
    |  Everything else  |
    * ----------------- */
 
+  private def executeSqlScript(
+      script: CompoundBody,
+      args: Map[String, Expression] = Map.empty): DataFrame = {
+    val sse = new SqlScriptingExecution(script, this, args)
+    var df: DataFrame = null
```

Comment:
why do we need this `var`?

Reply:
`SqlScriptExecution` next method returns DataFrame. We can remove this var and use `val` inside while loop.